com.unity.multiplayer.mlapi/Editor/CodeGen/XXHash/XXHash.cs

```diff
@@ -0,0 +1,291 @@
+// <copyright file="XXHash.cs" company="Sedat Kapanoglu">
+// Copyright (c) 2015-2019 Sedat Kapanoglu
+// MIT License (see LICENSE file for details)
+// </copyright>
+
+// @mfatihmar (Unity): Modified for Unity support
+
+using System.Runtime.CompilerServices;
+
+namespace MLAPI.Editor.CodeGen
+{
+    /// <summary>
+    /// XXHash implementation.
+    /// </summary>
+    internal static class XXHash
```

Comment:
Do we really need another hash method?

The MLAPI already has Fowler–Noll–Vo which it uses for everything internally.

Reply:
Maybe we don't need another hash method but we probably need xxHash instead of FNV.

xxHash is very efficient, well-known, highly-adopted non-cryptographic hashing algorithm.
You could see that over RPC RFC in the "Cross Compatibility" section but let me link those here too for you:
Section in the RFC: https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/blob/rfc/std-rpc-api/text/0000-std-rpc-api.md#cross-compatibility
xxHash website: https://cyan4973.github.io/xxHash/

Also we use xxHash in other places around Unity (e.g. DOTS Entities uses xxHash). I think xxHash excels in all aspects compared to our MLAPI implementation of FNV.

I'm happy to have another PR that replaces FNV with xxHash around the whole MLAPI codebase but for now I think I won't be doing that in this PR since xxHash will be used in the Editor only for now (other parts of hashing would touch runtime as well, so let's have a separate PR and discussion just for that).

Reply:
Ok, if nothing else, let's put in some verbiage on how we made this choice as well as something like "xxHash is also used by DOTs, chore task to remove this when we can import from standard Unity, etc."